### PR TITLE
PCHR-3435: Cleanup jQuery related partials

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
In this PR jquery related partials has been organised.

## Technical Details
1. Created "jquery" folder in the level of "bootstrap".
   - overrides
     - `civicrm/common/_datatables.scss` has been moved here
     - `civicrm/common/_bootstrap-calendar.scss` and `civicrm/common/_datepickers.scss` has been combined into `_ui-datepicker.scss` and moved here
     - `civicrm/common/_ui-dialog.scss` and `civicrm/common/_modals.scss` has been combined into `_ui-dialog.scss` and moved here.
     - `civicrm/common/_tabs.scss` has been moved here

2. Fixed linting errors in all touched files.
3. The following code has been completely removed
```scss
#bootstrap-theme .input-group-addon.fa-calendar { 
  background: none !important; 
  border: solid 1px $gray-dark !important; 
  border-left: 0 !important; 
  color: $brand-primary; 
  cursor: pointer; 
}
```
This piece of code was added part of https://github.com/compucorp/org.civicrm.shoreditch/commit/32916055a7eabc49da0e9b0d16a112d508ce6d20.
From the ticket in JIRA, all the related pages has been checked but this selector is not found, all its not found in Code. And Backstop did not show any errors.
4. The following code has been moved from jquery overrides to bootstrap overrides. Because only when in context of bootstrap, the override is necessary.
```scss
.ui-dialog-content#bootstrap-theme {
  background: $crm-white;
  padding: 10px 20px;
}
```
5. Moved the contents of `.ui-dialog` to a new mixin under `jquery/mixins/_ui-dialog.scss`. And reused it in `jquery/overrides/_ui-dialog.scss` and `civicrm/common/_modals.scss`.

## Tests
Backstop has been run for CiviHR, Mosaico and Civicrm and no errors are reported. And CiviCase has been tested manually.

### CODE REVIEW

File: scss/jquery/overrides/_ui-dialog.scss [hunk 1]
```diff
@@ -0,0 +1,238 @@
+@import 'SCSSROOT/org.civicrm.shoreditch/base/scss/vendor/bootstrap/modals';
+@import 'SCSSROOT/org.civicrm.shoreditch/scss/bootstrap/overrides/style/modals';
+
+.crm-container.ui-dialog {
```
Comment (AkA84) [inline scss/jquery/overrides/_ui-dialog.scss]: there should be no reference of `.crm-container` in this partial

File: scss/jquery/overrides/_ui-datepicker.scss [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      border-top: 1px solid rgb(221, 221, 221);
+    }
+
+    th {
+      color: $datepicker-gray-darker;
+      font-size: 11px;
+    }
+  }
+
+  [data-handler],
```
Comment (AkA84) [inline scss/jquery/overrides/_ui-datepicker.scss]: this looks weird, what's this for?

**Review by AkA84 — CHANGES_REQUESTED** [review]

File: scss/jquery/overrides/_ui-datepicker.scss (hunk 2 above)
Comment (deb1990) [reply scss/jquery/overrides/_ui-datepicker.scss]: This for the dates which are not disabled, and they dont have a class
2018-03-19 at 4 39 pm

Comment (AkA84) [reply scss/jquery/overrides/_ui-datepicker.scss]: leave a comment explaining that then, please

File: scss/jquery/overrides/_ui-dialog.scss (hunk 1 above)
Comment (deb1990) [reply scss/jquery/overrides/_ui-dialog.scss]: As discussed, moved the content of `.ui-dialog` to a new mixin under `jquery/mixins/_ui-dialog.scss`. And reused it in `jquery/overrides/_ui-dialog.scss` and `civicrm/common/_modals.scss`.

**Review by AkA84 — APPROVED** [review]

**Pull request merged**